src/Database/Redis/Connection.hs

```diff
@@ -212,6 +249,133 @@ newtype ClusterConnectError = ClusterConnectError Reply
 
 instance Exception ClusterConnectError
 
+data MaybeException = MaybeException String deriving (Show)
+
+instance Exception MaybeException
+
+data GoogleOAuthAccessTokenResponse =
```

Comment:
should we move all authToken refresh logic in separate file?
Just to split the code.

Reply:
Yes, Done